benchadapt/python/benchadapt/machine_info.py

```diff
@@ -77,6 +77,31 @@ class GitParseWarning(RuntimeWarning):
 
 
 def github_info():
+    """Get github metadata from environment variables"""
+    repo = os.environ.get("BENCHMARKABLE_REPOSITORY")
+    pr_number = os.environ.get("BENCHMARKABLE_PR_NUMBER")
+    commit = os.environ.get("BENCHMARKABLE_COMMIT")
+
+    gh_info = {
+        "commit": commit,
+        "repository": repo,
+        # "branch": None,
+        "pr_number": pr_number,
+    }
+
+    if not (repo and pr_number and commit):
+        warnings.warn(
+            (
+                "All of BENCHMARKABLE_REPOSITORY, BENCHMARKABLE_PR_NUMBER, and "
+                "BENCHMARKABLE_COMMIT must be set if `github` is not specified. "
+                f"\nValues: `{gh_info}`"
+            ),
+        )
```

Comment:
Should this be an error or a warning? Or is it something we want as a warning for now while we transition and will become an error? 

The docs below seem to suggest it should be an error

```
        By default, metadata will be obtained from ``BENCHMARKABLE_REPOSITORY``,
        ``BENCHMARKABLE_COMMIT``, and ``BENCHMARKABLE_PR_NUMBER`` environment variables.
        If any are unset, an error will be raised.
```

Reply:
Oops, I forgot to update when I changed it from an error to a warning; fixed now. An error would be nice, but that causes workflow problems because the error will get raised before the field can get overridden in adapters, so I changed it to a warning.